core/app/models/spree/promotion.rb

```diff
@@ -257,10 +257,9 @@ def match_all?
       match_policy == "all"
     end
 
-    def apply_automatically_disallowed_with_codes_or_paths
+    def apply_automatically_disallowed_with_paths
       return unless apply_automatically
 
-      errors.add(:apply_automatically, :disallowed_with_code) if codes.any?
```

Comment:
Can you please expand a little bit more in its commit description? I'm not sure to get why this is needed.

Reply:
I've added a short message to explain why this commit is needed, let me know if is clear enough

Reply:
@kennyadsl did you read the new commit message? Let me know what do you think

Reply:
Thanks for updating it, I think the second part can be rewritten as:

```
Also, this improves backward compatibility since if we keep validating
against promotions with codes as well, it will be impossible to update 
existing apply-auto promotions that already have a code associated.
```

While in the first part I still don't get this sentence:

```
Since promotions creation is a two-step process and promo codes
are added in the second step, the validation have to be there.
```

Why do you say that the validation has to be there if we are removing it in this commit?

Reply:
I modified the commit description to explain why we need to remove that validation and move it in the promo code creation step, let me know if now is ok